Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java

```diff
@@ -164,7 +166,7 @@ private Uri buildStrongMatchUrl(String matchDomain, DeviceInfo deviceInfo, PrefH
             String hardwareIDTypeVal = deviceInfo.isHardwareIDReal() ? Defines.Jsonkey.HardwareIDTypeVendor.getKey() : Defines.Jsonkey.HardwareIDTypeRandom.getKey();
             uriString += "&" + Defines.Jsonkey.HardwareIDType.getKey() + "=" + hardwareIDTypeVal;
             // Add GAID if available
-            if (systemObserver.GAIDString_ != null) {
+            if (systemObserver.GAIDString_ != null && !BranchUtil.isTestModeEnabled(context_)) {
```

Comment:
Please pass `context` to `buildStrongMatchUrl()` method instead of creating a class member variable